examples/flaskr/flaskr/auth/models.py

```diff
@@ -1,15 +1,20 @@
+from typing import TYPE_CHECKING
+
+from sqlalchemy.orm import Mapped, mapped_column, relationship
 from werkzeug.security import check_password_hash
 from werkzeug.security import generate_password_hash
 
 from flaskr import db
 
+if TYPE_CHECKING:
+    from flaskr.blog.models import Post
 
-class User(db.Model):
-    id = db.Column(db.Integer, primary_key=True)
-    username = db.Column(db.String, unique=True, nullable=False)
-    password_hash = db.Column(db.String, nullable=False)
 
-    posts = db.relationship("Post", back_populates="author")
+class User(db.Model):
+    id: Mapped[int] = mapped_column(primary_key=True)
+    username: Mapped[str] = mapped_column(unique=True, nullable=False)
+    password_hash: Mapped[str] = mapped_column(nullable=False)
```

Comment:
ugh i can't believe that SA didn't add something nicer for the most likely very common case of "primitive type, but not nullable"...

Reply:
`Mapped[str]` does mean `nullable=False`, and `Mapped[str | None]` means `nullable=True`, so we can remove the `mapped_column` for some of these.